pages/_app.tsx

```diff
@@ -1,20 +1,55 @@
 import "../styles/globals.css";
-import React from "react";
+import React, { useEffect } from "react";
 import type { AppProps } from "next/app";
+import { useRouter } from "next/router";
+import Script from "next/script";
 import { ThemeProvider } from "@emotion/react";
 import styled from "@emotion/styled";
 import { QueryClient, QueryClientProvider } from "react-query";
 import { ReactQueryDevtools } from "react-query/devtools";
 import { RecoilRoot } from "recoil";
 import { Layout } from "~/components/uis";
 import { MemberInfo } from "~/contexts";
+import { GA_TRACKING_ID, gtagPageView } from "~/libs/gtag";
 import { emotionTheme } from "~/theme";
 
 const queryClient = new QueryClient();
 
 function MyApp({ Component, pageProps }: AppProps) {
+  const router = useRouter();
+  useEffect(() => {
+    const handleRouteChange = (url: string) => {
+      gtagPageView(url);
+    };
+    router.events.on("routeChangeComplete", handleRouteChange);
+    router.events.on("hashChangeComplete", handleRouteChange);
+
+    return () => {
+      router.events.off("routeChangeComplete", handleRouteChange);
+      router.events.off("hashChangeComplete", handleRouteChange);
+    };
+  }, [router.events]);
```

Comment:
router.events가 변경되면 handleRouteChange를 이벤트핸들러로 한번만 등록하는 걸 보장할 수 있을 지 걱정이 되네

마운트시에 router.events가 있는게 보장된다면 아래처럼 하면 어떨까?

```suggestion
  useEffect(() => {
    const handleRouteChange = (url: string) => {
      gtagPageView(url);
    };
    router.events.on("routeChangeComplete", handleRouteChange);
    router.events.on("hashChangeComplete", handleRouteChange);

    return () => {
      router.events.off("routeChangeComplete", handleRouteChange);
      router.events.off("hashChangeComplete", handleRouteChange);
    };
  }, []);
```

Reply:
굿굿 테스트 했을 때 마운트시에 `router.events` 가 있는 것 같아서 commit suggestion 할게~